gooey/gui/components/widgets/core/chooser.py

```diff
@@ -64,40 +64,57 @@ def getValue(self):
 class FileChooser(Chooser):
     """ Retrieve an existing file from the system """
     def getDialog(self):
-        return wx.FileDialog(self, _('open_file'), style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST)
+        options = self.Parent._options
+        return wx.FileDialog(self, message=options.get('message', _('open_file')),
+                             style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+                             defaultFile=options.get('defaultFile', _("enter_filename")),
+                             defaultDir=options.get('defaultDir', _('')),
+                             wildcard=options.get('wildcard', wx.FileSelectorDefaultWildcardStr))
 
 
 class MultiFileChooser(Chooser):
     """ Retrieve an multiple files from the system """
     def getDialog(self):
-        return wx.FileDialog(self, _('open_files'), style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST | wx.FD_MULTIPLE)
+        options = self.Parent._options
+        return wx.FileDialog(self, message=options.get('message', _('open_files')),
+                             style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST | wx.FD_MULTIPLE,
+                             defaultFile=options.get('defaultFile', _("enter_filename")),
+                             defaultDir=options.get('defaultDir', _('')),
+                             wildcard=options.get('wildcard', wx.FileSelectorDefaultWildcardStr))
+
     def getResult(self, dialog):
         return os.pathsep.join(dialog.GetPaths()) 
 
 
 class FileSaver(Chooser):
     """ Specify the path to save a new file """
     def getDialog(self):
+        options = self.Parent._options
         return wx.FileDialog(
             self,
             style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
-            defaultFile=_("enter_filename"),
-            message=_('choose_file')
+            defaultFile=options.get('defaultFile', _("enter_filename")),

```

Comment:
These option keys should actually be snake_cased as they come from Gooey's public options API, which honors PEP8.

Reply:
hi
i kept options as they appear in original wx lib so users will have the same interface in Gooey as they have in wx.
besides - its a string dict key, so i dont think pep8 has any case enforce on this
(make sure you are not confused by the removed code which had variables and they really should have been snake_case)

i think its smarter to have it the same as wx, but if you feel like it should be snake_case i will change that.

Reply:
As much as possible, we shouldn't be exposing wx internals to outside clients or using it as a rational for driving APIs. Yes, we happen to be using the same keyname as wx here. Their names are Good Enough that it doesn't matter -- `default_dir` has no coupling to wx specifically -- we could rewrite Gooey with Tkinter or PyQt and use the reuse the name. Additionally, your options would be the _only_ options in Gooey's public API that are camelCased. That alone should be reason enough to snake_case.